File: First Unique Character in a String (Easy, #387)/FirstUnique.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        mydict = dict()
        index = 0

        for c in s:
            if c not in mydict.keys():
                mydict[c] = index
            else:
                mydict[c] = -1
            index += 1
        
        for value in mydict.values():
            if value != -1:
                return value

        return -1

# Time complexity O(n) and Space complexity O(1)

File: First Unique Character in a String (Easy, #387)/FirstUnique.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        map<char, int> my_map = {};
        int index = 0;

        for(char c : s) {
            if (my_map.find(c) == my_map.end()) {
                my_map[c] = index;
            }
            else {
                my_map[c] = -1;
            }
            ++index;
        }

        for(char c : s) {
            if (my_map[c] != -1)
                return my_map[c];
        }

        return -1;
    }
};

// Time complexity O(n) and Space complexity O(1)